etl/datadiff.py

```diff
@@ -261,6 +269,14 @@ def _append_and_print(x):
     exit(1 if any_diff else 0)
 
 
+def _index_equals(table_a: pd.DataFrame, table_b: pd.DataFrame, sample: int = 1000) -> bool:
+    """Check if two tables have the same index. Sample both tables to speed up the check."""
+    return (
+        table_a.sample(min(len(table_a), sample), random_state=0).index
+        == table_b.sample(min(len(table_b), sample), random_state=0).index
+    ).all()  # type: ignore
```

Comment:
Interesting idea. I'm a bit surprised that this would be quicker than just comparing `(table_a.index == table_b.index).all()` or `table_a.index.equals(table_b.index)`. But I suppose you have checked and for big tables sampling is better.
Maybe you could include
```
if (len(table_a) < sample):
    return table_a.index.equals(table_b.index)
else:
    return (...).all()
```

Reply:
This optimisation was done for `ghe` where it speeds it up from almost 2 minutes to 4 seconds. It has huge multi-level index which is probably why it takes so long.

Thanks, used your approach.